From 5946549795c5403f47b992335bdb3969eba08b6a Mon Sep 17 00:00:00 2001
From: Bart Ribbers <bribbers@disroot.org>
Date: Sun, 18 Jan 2026 21:32:56 +0100
Subject: [PATCH] sddm-wayland-session: install to KDE_INSTALL_LIBDIR

According to https://github.com/sddm/sddm/blob/63780fcd79f1dbf81a30eef48c28c699ab15aded/src/common/ConfigReader.cpp#L146
config files can be installed to `/usr/lib/sddm/sddm.conf.d` which is a
better location than anywhere in `/etc` as us shipping this makes it a
system setting rather than a user setting.

This makes it easier for distributions to ship potential updates to this
file, especially when the user modified it for themselves, as they don't
have to merge any changes and can just replace it. Any user modification
will remain active and of an higher priority.
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a14dbf797a..a985bdb3e3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -391,7 +391,7 @@ add_subdirectory(login-sessions)
 add_subdirectory(lookandfeel)
 
 if (INSTALL_SDDM_WAYLAND_SESSION)
-    install(FILES sddm-wayland-session/plasma-wayland.conf DESTINATION /etc/sddm.conf.d)
+    install(FILES sddm-wayland-session/plasma-wayland.conf DESTINATION ${KDE_INSTALL_LIBDIR}/sddm/sddm.conf.d)
 else()
     message(STATUS "INSTALL_SDDM_WAYLAND_SESSION is disabled. As soon as it's installed, SDDM will default to use Wayland and KWin for its greeter session (BETA, do not deploy to final users yet).")
 endif()
-- 
2.52.0

